api/envoy/api/v2/route/route.proto

```diff
@@ -358,6 +358,14 @@ message RouteAction {
   // Indicates that during forwarding, the matched prefix (or path) should be
   // swapped with this value. This option allows application URLs to be rooted
   // at a different path from those exposed at the reverse proxy layer.
+  //
+  // .. attention::
+  //
+  //   Pay careful attention to the use of trailing slashes in the
+  //   :ref:`route's match <envoy_api_field_route.Route.match>` prefix value.
+  //   Stripping a prefix from a path requires multiple Routes to handle all cases. For example,
+  //   rewriting */prefix* to */* and */prefix/etc* to */etc*
+  //   cannot be done in a single :ref:`Route <envoy_api_msg_route.Route>`.
```

Comment:
Can we give an explicit example of what the user would configure? I think that would be helpful.

Also, Q: I lost track of the conversation but isn't there a targeted fix possible here where we just adjust/normalize the slash where the prefix ends and the rest of the route begins?

Reply:
> Can we give an explicit example of what the user would configure? I think that would be helpful.

I added an example in my previous commit, https://github.com/dio/envoy/blob/d46c26e4ede389971fbfeca7bd2d2e48716ea2d4/api/envoy/api/v2/route/route.proto#L369-L393 if you think this is good enough I can add it back.

Reply:
> Also, Q: I lost track of the conversation but isn't there a targeted fix possible here where we just adjust/normalize the slash where the prefix ends and the rest of the route begins?

Reply:
Yeah my thinking here is that slashes have no special role in the URL after the first 3.  E.g.
  scheme://origin/foo/bar
and
  scheme://origin/foo//bar
are both different valid URLs and origin servers can interpret them differently.  Moreover, proxies may see arbitrary slashes in URL parameters.  If they originate from a browser they'd be escaped, but not if they originate from some other client. 

You could normalize the match strings, but you only lose generality doing so, and it's conceivable someone may want a URL rewriter that strips off a prefix that does not necessarily end with a slash.

Authoritative spec is https://www.w3.org/Addressing/URL/url-spec.txt which has this (IMO) somewhat ambiguous statement:
>   The path is interpreted in a manner dependent on the scheme being
   used. Generally, the reserved slash  "/" character (ASCII 2F hex)
   denotes a level in a hierarchical structure, the higher level part
   to the left of the slash. 

So in typical use (e.g. mapping to a file-system) might correspond to a file-system where normalization makes sense, that's not *required* by the spec and in practice I think it's better not to mess with URLs too much.

Reply:
I think what I'm suggesting is to add an explicit example above in the docs where you talk about how it cannot be done in a single route rule. Can you list out the two rules that cover the example case so the user better understands?

Reply:
Thank you @dio! Very clear.